libminifi/test/integration/C2MetricsTest.cpp

```diff
@@ -131,23 +134,67 @@ class MetricsHandler: public HeartbeatHandler {
       runtime_metrics.HasMember("flowInfo") &&
       runtime_metrics["flowInfo"].HasMember("versionedFlowSnapshotURI") &&
       runtime_metrics["flowInfo"].HasMember("queues") &&
-      runtime_metrics["flowInfo"].HasMember("components") &&
       runtime_metrics["flowInfo"]["queues"].HasMember("2438e3c8-015a-1000-79ca-83af40ec1997") &&
-      runtime_metrics["flowInfo"]["components"].HasMember("FlowController") &&
-      runtime_metrics["flowInfo"]["components"].HasMember("GetTCP") &&
-      runtime_metrics["flowInfo"]["components"].HasMember("LogAttribute");
+      runtime_metrics["flowInfo"].HasMember("processorStatuses") &&
+      [&]() {
+        if (runtime_metrics["flowInfo"]["processorStatuses"].GetArray().Size() != 2) {
+          return false;
+        }
+        for (const auto& processor : runtime_metrics["flowInfo"]["processorStatuses"].GetArray()) {  // NOLINT(readability-use-anyofallof)
```

Comment:
I side with the linter here, we should use `all_of`. The predicate callback can still throw if necessary.

Reply:
Updated in https://github.com/apache/nifi-minifi-cpp/pull/1909/commits/2723bace043147136941c68f86216fcd61d74a36